test/verify/check-apps

```diff
@@ -0,0 +1,150 @@
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import parent
+from testlib import *
+from packagelib import *
+
+@skipImage("Not supported", "continuous-atomic", "fedora-atomic", "rhel-atomic")
+class TestApps(PackageCase):
+
+    def setUp(self):
+        PackageCase.setUp(self)
+        self.appstream_collection = set()
+
+    def createAppStreamPackage(self, name, version, revision):
+        self.createPackage(name, version, revision, content = {
+            "/usr/share/metainfo/org.cockpit-project.{0}.metainfo.xml".format(name): """
+<component type="addon">
+  <extends>cockpit.desktop</extends>
+  <id>org.cockpit-project.{0}</id>
+  <name>{0}</name>
+  <summary>An application for testing</summary>
+            <launchable type="cockpit-manifest">{0}</launchable>
+</component>
+""".format(name) })
+        self.appstream_collection.add(name)
+
+    def enableAppStreamRepo(self):
+        body = ""
+        for p in self.appstream_collection:
+            body += """
+  <component type="addon">
+    <extends>cockpit.desktop</extends>
+    <id>org.cockpit-project.{0}</id>
+    <name>{0}</name>
+    <summary>An application for testing</summary>
+    <launchable type="cockpit-manifest">{0}</launchable>
+    <pkgname>{0}</pkgname>
+  </component>
+""".format(p)
+        self.machine.execute("mkdir -p /usr/share/app-info/xmls")
+        self.createPackage("appstream-data-test", "1.0", "1", content = {
+            "/usr/share/app-info/xmls/test.xml": """
+<components origin="test">
+{0}
+</components>
+            """.format(body)})
+        self.enableRepo()
+        self.machine.execute("pkcon refresh force")
+        self.machine.execute("pkcon -y install appstream-data-test")
+
+    def testBasic(self):
+        b = self.browser
+        m = self.machine
+
+        # Make sure none of the appstream directories exist.  They
+        # will be created later and we need to cope with that.
+
+        m.execute("rm -rf /usr/share/metainfo /usr/share/app-info /var/cache/app-info")
+
+        m.start_cockpit()
+        b.login_and_go("/apps")
+        b.wait_present(".app-list-empty")
+
+        self.createAppStreamPackage("app-1", "1.0", "1")
+        self.enableAppStreamRepo()
+
+        b.wait_present(".app-list tr:contains('app-1') button:contains('Install')")
+        b.click(".app-list tr:contains('app-1') button:contains('Install')")
```

Comment:
Could you add some rpm query here to ensure that the package actually got installed?

Reply:
I have added a `test -f /stamp-...` test, like in `check-packagekit`.  Did you have something else in mind?

Reply:
That's what I had in mind indeed, thanks! (Also a corresponding `! test -f` after removal).